### ISSUE TITLE
Externalise strings

### ISSUE THREAD

**Issue by QuiCM** [NyxStudios/Orion#39]
Over the years we have had many requests for TShock to be translated, as well as offers from translators to do so. However, due to the nature of the codebase (i.e., hardcoded strings), this is a difficult task.

I propose we make use of Visual Studio's built in resx file support (read more [here](https://msdn.microsoft.com/en-us/library/7k989cfy%28v=vs.90%29.aspx)) to help remedy this issue.

**Comment by QuiCM** [comment]
@NyxStudios/tshock Initial work for this issue has been submitted on the [external-strings](https://github.com/NyxStudios/Orion/tree/external-strings) branch. Feedback would be appreciated

**Comment by hakusaro** [comment]
I'm generally neutral/abstaining on this issue; I think that _any type of externalization is better than nothing_ but I personally don't know enough about the resx format to know the pros & cons of choosing it over say, yaml.

**Comment by QuiCM** [comment]
Resx makes it ridiculously easy to add/modify strings.
Visual Studio automatically creates a backing `Resource.Designer.cs` class that updates with fields as you add strings to your resx file. Strings are then accessed through `namespace.Resource.{field}`. 
In our case I've renamed the resx file to `Language.resx`, so to access the text displayed when loading a plugin or service you would use `var str = Language.LoadingPluginOrService;` (with a `using Orion.Languages` to access the Language class).

When the program is compiled, VS runs resgen.exe which compiles the .resx files into .resources files (which are then embedded in the output assembly).

Furthermore, a program using resx files intelligently chooses which language to use based on the system culture and the available files. In the external-strings branch I have added `Language.es.resx` - on a system using a Spanish language pack, those strings will be used instead of the default `Language.resx`. Any other languages will fallback to the default English translation unless a translation is provided for them.

I have no idea what kind of features can be provided with yaml/json/other plain text formats, but I'm fairly certain they provide fewer features and ease of use out of the box.
